### PR TITLE
Make construct chaining configurable

### PR DESCRIPTION
**Pull request by PizieDust**
This PR provides a configurable option for construct chaining. 
The default behavior let's construct automatically compute values for the next typed hole. This option can be used to turn this behavior off.

```json
"ocaml.server.construct.chainConstruct": true
```

image

cc @voodoos

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # Unreleased
 
+- Add `ocaml.server.construct.chainConstruct` setting to configure construct chaining. (#1673)
```
Comment (voodoos) [inline CHANGELOG.md:5]: I think we could have a new group to distinguish options that are used to configure the server and options that are used to configure the client's commands behavior. Maybe `ocaml.commands.construct.recursiveCalls`

File: package.json [hunk 2]
```diff
@@ -306,6 +306,11 @@
           "default": false,
           "markdownDescription": "Enable/Disable syntax documentation"
         },
+        "ocaml.server.construct.chainConstruct": {
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Enable/Disable chaining construct to display values for subsequent holes."
```
Comment (voodoos) [inline package.json:312]:
```suggestion
          "markdownDescription": "When enabled, the construct command will execute again on the next hole after a value has been chosen."
```

File: vscode.opam [hunk 3]
```diff
@@ -22,7 +22,7 @@ depends: [
   "gen_js_api" {= "1.1.2"}
   "promise_jsoo" {>= "0.3.1"}
   "jsonoo" {>= "0.2.1"}
-  "ocamlformat" {= "0.26.2" & with-dev-setup}
+  "ocamlformat" {= "0.27.0" & with-dev-setup}
```
Comment (voodoos) [inline vscode.opam:25]: Why is this part of this PR ? I think you should rebase on master, the bump was made recently.

**Review by voodoos — APPROVED** [review]
Thanks, apart from some nitpicking on the name choices this looks good to me.

File: vscode.opam (hunk 3 above)
Comment (smorimoto) [reply vscode.opam]: Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (voodoos) [reply CHANGELOG.md:5]: @PizieDust did you forget this one ? After that I think the PR is ready to be merged :-)

Comment (PizieDust) [reply CHANGELOG.md:5]: Oh my bad. let me fix that

**Review by voodoos — APPROVED** [review]
Thanks, looks good to me

**Pull request merged**